### PR TITLE
CI: drop Python 3.7 & add 3.10 for `lama-to-dask` workflow

### PR DESCRIPTION
**Pull request by sadielbartholomew**
For context, see https://github.com/NCAS-CMS/cf-python/pull/313#issuecomment-1033476035 and related comments in that thread. Also adding Python 3.10 whilst I am amending the workflow matrix, as that is now available and it is a good idea to start testing with it.

Trivial change but opening as a PR instead of committing directly to the branch in order to check the remaining jobs pass. (`lama` was changed to Llama somehow when writing the commit message here and it can stay for amusement value / because I can't be bothered to rebase to re-word).

### CODE REVIEW

**Comment by sadielbartholomew** [comment]
Ah, `3.10` is being interpreted as `3.1` naturally, let me try again...

**Comment by sadielbartholomew** [comment]
Retriggering via open-close...

**Comment by sadielbartholomew** [comment]
Oh, fair, the Python 3.10 jobs (empirically) won't pass because something about `cfdm` doesn't seem to play ball with a 3.10 update (yet to investigated):


```python
Traceback (most recent call last):
  File "/Users/runner/work/cf-python/cf-python/main/cf/test/test_Data.py", line 23, in <module>
    import cf
  File "/Users/runner/work/cf-python/cf-python/main/cf/__init__.py", line 87, in <module>
    import cfdm
  File "/Users/runner/work/cf-python/cf-python/cfdm/cfdm/__init__.py", line 44, in <module>
    from . import core
  File "/Users/runner/work/cf-python/cf-python/cfdm/cfdm/core/__init__.py", line 60, in <module>
    from .domain import Domain
  File "/Users/runner/work/cf-python/cf-python/cfdm/cfdm/core/domain.py", line 4, in <module>
    class Domain(mixin.FieldDomain, abstract.Properties):
  File "/Users/runner/work/cf-python/cf-python/cfdm/cfdm/core/meta/docstringrewrite.py", line [20](https://github.com/NCAS-CMS/cf-python/runs/5130994352?check_suite_focus=true#step:13:20)5, in __new__
    wrapper.__wrapped__ = attr
AttributeError: readonly attribute
Error: Process completed with exit code 1.
```

so maybe that was a bit too optimistic. Removing 3.10 from the workflow, we'll just run on 3.8 and 3.9 for now...

**Comment by sadielbartholomew** [comment]
Should just be the 3.8 and 3.9 jobs now, retriggering...

**Comment by sadielbartholomew** [comment]
All good to merge now.

**Comment by davidhassell** [comment]
Thanks, Sadie.
